storage/imx6ull-flash/flashdrv.c

```diff
@@ -447,16 +447,20 @@ static void flashdrv_setDevClock(int dev, int state)
 flashdrv_dma_t *flashdrv_dmanew(void)
 {
 	flashdrv_dma_t *dma = mmap(NULL, _PAGE_SIZE, PROT_READ | PROT_WRITE, MAP_UNCACHED, OID_NULL, 0);
-	dma->last = NULL;
-	dma->first = NULL;
+	if (dma != MAP_FAILED) {
+		dma->last = NULL;
+		dma->first = NULL;
+	}
 
 	return dma;
 }
 
 
 void flashdrv_dmadestroy(flashdrv_dma_t *dma)
 {
-	munmap(dma, _PAGE_SIZE);
+	if (dma != MAP_FAILED && dma != NULL) {
+		munmap(dma, _PAGE_SIZE);
+	}
```

Comment:
When can dma be null? Also I beleive that check for dma == MAP_FAILED is redundant as this should be checked just after performing mmap.

Reply:
Take a look at `setup_flash_info()` for MAP_FAILED case. NULL check is probably added as protection against careless deinitialization.